submariner/OVN-Interconnect.md

```diff
@@ -0,0 +1,244 @@
+# Submariner Enhancement for OVN Interconnect
+
+<!-- Add link to issue/epic if available -->
+
+## Summary
+
+OVN CNI is moving to a new network topology powered by OVN Interconnect Feature. This allows independent OVN deployments,
+within a kubernetes cluster, to be interconnected by OVN-managed GENEVE tunnels. Submariner does not work with these changes
+out of the box and this proposal explains the changes required to support it.
+
+## Proposal
+
+With OVN Interconnect we can have two types of deployment
+
+* Single Zone(Global Zone): A single-zone deployment will have only one OVN database and a set of master nodes programming it. This is similar
+  to the topology that we use now. But it will have a single `global` zone assigned to it with a transit switch which is not involved in any
+  packet forwarding. Though this setup is expected to work by default, the subnet that the transit switch uses overlaps with the subnet
+  Submariner uses for the logical router. In this proposal, we plan to remove the submariner logical router and the associated switches and
+  this will not be a problem anymore.
+
+  The below indicates node annotations in a single zone OVN setup and is added by OVN.
+
+```bash
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.1/16"]'
+      k8s.ovn.org/ovn-zone: global
+    name: cluster1-worker
+
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.1/16"]'
+      k8s.ovn.org/ovn-zone: global
+    name: cluster2-worker
+```
+
+* Multiple Zone: In a multiple-zone setup, we will have an OVN database and an OVN master pod for each zone. Transit switches connect the
+  zones. The OVN-Kubernetes services ensure that the necessary routes are added for pod and service reachability across nodes in different
+  zones.
+
+```bash
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.3/16"]'
+      k8s.ovn.org/ovn-zone: global
+    name: cluster1-worker
+
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.5/16"]'
+      k8s.ovn.org/ovn-zone: az2
+    name: cluster2-worker
+```
+
+With the current architecture, Submariner adds routes only in the zone in which it is deployed. For example, if Submariner is deployed in
```

Comment:
This sentence gives an impression like Submariner can be installed only in a zone. When we install Submariner in a Cluster (even with OVN-IC enabled) couple of Submariner pods will run on each and every node of the cluster, but may not be aware of the zone knowledge in the current architecture. So we should clarify that the network-plugin-syncer will only program a single zone and pods in other zones will not be able to communicate with remote clusters.

Reply:
Done